core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -1175,7 +1177,7 @@ private[deploy] class Master(
   }
 
   private def newDriverId(submitDate: Date): String = {
-    val appId = "driver-%s-%04d".format(createDateFormat.format(submitDate), nextDriverNumber)
+    val appId = driverIdPattern.format(createDateFormat.format(submitDate), nextDriverNumber)
```

Comment:
I guess we need to handle exceptions here after this change.

Reply:
This is addressed via `checkValue`.